identity_iota_core/packages/iota_identity/sources/controller.move

```diff
@@ -0,0 +1,143 @@
+module iota_identity::controller {
+  use iota::transfer::Receiving;
+  use iota::borrow::{Self, Referent, Borrow};
+  use iota_identity::permissions;  
+
+  public use fun delete_controller_cap as ControllerCap.delete;
+  public use fun delete_delegation_token as DelegationToken.delete;
+
+  /// This `ControllerCap` cannot delegate access.
+  const ECannotDelegate: u64 = 0;
+  // The permission of the provided `DelegationToken` are not
+  // valid to perform this operation.
+  const EInvalidPermissions: u64 = 1;
+
+  /// Event that is created when a new `DelegationToken` is minted.
+  public struct NewDelegationTokenEvent has copy, drop {
+    controller: ID,
+    token: ID,
+    permissions: u32,
+  }
+
+  /// Capability that allows to access mutative APIs of a `Multicontroller`.
+  public struct ControllerCap has key {
+    id: UID,
+    can_delegate: bool,
+    access_token: Referent<DelegationToken>,
+  }
+
+  public fun id(self: &ControllerCap): &UID {
+    &self.id
+  }
+
+  /// Borrows this `ControllerCap`'s access token.
+  public fun borrow(self: &mut ControllerCap): (DelegationToken, Borrow) {
+    self.access_token.borrow()
+  }
+
+  /// Returns the borrowed access token together with the hot potato.
+  public fun put_back(self: &mut ControllerCap, token: DelegationToken, borrow: Borrow) {
+    self.access_token.put_back(token, borrow);
+  }
+
+  /// Creates a delegation token for this controller. The created `DelegationToken`
+  /// will have full permissions. Use `delegate_with_permissions` to set or unset
+  /// specific permissions.
+  public fun delegate(self: &ControllerCap, ctx: &mut TxContext): DelegationToken {
+    new_delegation_token(self.id.to_inner(), permissions::all(), ctx)
```

Comment:
I guess, I missed some context, so sorry for the stupid question. ^^;;

Are we missing the `can_delegate` check, here? Just wondering, as it's done in the next function.

```suggestion
    assert!(self.can_delegate, ECannotDelegate);
    new_delegation_token(self.id.to_inner(), permissions::all(), ctx)
```